o2a/mappers/prepare_mixin.py

```diff
@@ -27,24 +29,25 @@ class PrepareMixin:
     def has_prepare(oozie_node):
         return bool(xml_utils.find_nodes_by_tag(oozie_node, "prepare"))
 
-    def get_prepare_command(self, oozie_node: ET.Element, params: Dict[str, str]):
-        # In BashOperator in Composer we can't read from $DAGS_FOLDER (~/dags) - permission denied.
-        # However we can read from ~/data -> /home/airflow/gcs/data.
-        # The easiest way to access it is using the $DAGS_FOLDER env variable.
-        delete_paths, mkdir_paths = self.parse_prepare_node(oozie_node, params)
-        if delete_paths or mkdir_paths:
+    def get_prepare_task(self, oozie_node: ET.Element, name: str, trigger_rule: str, property_set) -> Task:
+        delete_paths, mkdir_paths = self.parse_prepare_node(oozie_node, property_set=property_set)
+        delete = None
+        mkdir = None
+        if not delete_paths and not mkdir_paths:
+            raise Exception(f"There is neither delete nor mkdir in the {oozie_node}'s prepare.")
```

Comment:
In XSD schema, I do not see such a limitation. In automatically generated XML files, this element can therefore appear. I think this check should appear in the has_prepare method. What do you think?

Reply:
I actually simplified it even further. I reworked prepare_mixin to take oozie_node as __init__ parameter and the whole prepare functionality now is pretty much embedded in Prepare mixin.  See 40e3e4d4ec448c28f2ff4c60e5384ca20083ed23

Reply:
please resolve it @mik-laj if you think the discussion is resolved.